include/roboteam_ai/control/positionControl/CollisionDetector.h

```diff
@@ -0,0 +1,60 @@
+//
+// Created by ratoone on 10-12-19.
+//
+
+#ifndef RTT_COLLISIONDETECTOR_H
+#define RTT_COLLISIONDETECTOR_H
+
+#include <world/Field.h>
+#include <world_new/Robot.hpp>
+#include "control/ControlUtils.h"
+
+
+namespace rtt::ai::control{
+
+/**
+ * Checks for collision between points and different components of the field: robots, defence area, etc.
+ * Check isCollisionBetweenPoints method for more details
+ */
+class CollisionDetector {
+private:
+    const double DEFAULT_ROBOT_COLLISION_RADIUS = 3.0*Constants::ROBOT_RADIUS();
+
+    const std::vector<rtt::world_new::robot::Robot> &robots;
+    world::Field* field = nullptr;
+
+public:
+    /**
+     * Initialize the collision detector with the robot objects
+     * @param robots
+     */
+    explicit CollisionDetector(const std::vector<rtt::world_new::robot::Robot> &robots);
```

Comment:
Should be views

Reply:
Then, question: why does WorldData::getRobots() return a vector of Robots, instead of RobotViews? There isn't any other way to get all the robots, at least from what I've seen

Reply:
getRobotsNonOwning()

getRobots() is an O(1) operationg, getRobotsNonOwning is O(n)

Reply:
I'll open an issue with this and implement it after fixing that part of the world